frame.test.js

```diff
@@ -0,0 +1,160 @@
+const Frame = require('./frame')
+const Scorecard = require('./scorecard')
+
+describe('Frame', () => {
+  let frame
+  let scorecard
+
+  beforeEach(() => {
+    frame = new Frame()
+    scorecard = new Scorecard()
+  })
+
+  describe('a simple frame of two rolls with no strikes or spares', () => {
+    it('gives the score at the start of a frame', () => {
+      expect(frame.calculateTotalScore()).toBe(0)
+    })
+
+    it('captures score of first roll', () => {
+      frame.enterFirstRollScore(3, scorecard)
+      expect(frame.firstRollScore).toEqual(3)
+    })
+
+    it('adds first roll score to frame total', () => {
+      frame.enterFirstRollScore(3, scorecard)
+      expect(frame.calculateTotalScore()).toEqual(3)
+    })
+
+    it('captures score of second roll', () => {
+      frame.enterSecondRollScore(2, scorecard)
+      expect(frame.secondRollScore).toEqual(2)
+    })
+
+    // This is taken from the example of the first two rolls on example_ten_pin_scoring.png
+    it('knows a roll of one on the first roll and four on the second roll scores a total of five', () => {
+      frame.enterFirstRollScore(1, scorecard)
+      frame.enterSecondRollScore(4, scorecard)
+      expect(frame.calculateTotalScore()).toBe(5)
+    })
+  })
+
+  describe('when calculating any bonus scores', () => {
+    it('flags the frame as a strike when a strike is scored', () => {
+      frame.enterFirstRollScore(10, scorecard)
+      expect(frame.strikeBoolean).toBe(true)
+    })
+
+    it('flags the frame as a spare when a spare is scored', () => {
+      frame.enterFirstRollScore(6, scorecard)
+      frame.enterSecondRollScore(4, scorecard)
+      expect(frame.spareBoolean).toBe(true)
+    })
+
+    it('adds a bonus to the spare frame total score', () => {
+      frame2 = new Frame()
+      scorecard.frames = [frame, frame2]
+      frame.enterFirstRollScore(6, scorecard)
+      frame.enterSecondRollScore(4, scorecard)
+      frame2.enterFirstRollScore(5, scorecard)
+      expect(frame.bonusScore).toEqual(5)
+      expect(frame.totalScore).toEqual(15)
+    })
+
+    it('flags the frame as a strike when a strike is scored', () => {
+      frame.enterFirstRollScore(10, scorecard)
+      expect(frame.strikeBoolean).toBe(true)
+    })
+
+    it('adds both the current frame roll scores to the previous frame bonus', () => {
+      frame2 = new Frame()
+      scorecard.frames = [frame, frame2]
+      frame.enterFirstRollScore(10, scorecard)
+      frame2.enterFirstRollScore(5, scorecard)
+      frame2.enterSecondRollScore(4, scorecard)
+      expect(frame.bonusScore).toEqual(9)
```

Comment:
No expectation of total score so can not see if a strike is being implemented correctly to the final score.